digi/xbee/xsocket.py

```diff
@@ -0,0 +1,814 @@
+        response_packet = self.__xbee_device.send_packet_sync_and_get_response(option_packet, self.__timeout)
+        self.__check_response(response_packet)
+
+        # Return the option data.
+        return response_packet.option_data
+
+    def add_socket_state_callback(self, callback):
+        """
+        Adds a callback for the event :class:`digi.xbee.reader.SocketStateReceived`.
+
+        Args:
+            callback (Function): the callback. Receives two arguments.
+
+                * The socket ID as an Integer.
+                * The state received as a :class:`.SocketState`
+        """
+        self.__xbee_device.add_socket_state_received_callback(callback)
+
+    def del_socket_state_callback(self, callback):
+        """
+        Deletes a callback for the callback list of :class:`digi.xbee.reader.SocketStateReceived` event.
+
+        Args:
+            callback (Function): the callback to delete.
+
+        Raises:
+            ValueError: if ``callback`` is not in the callback list of
+                :class:`digi.xbee.reader.SocketStateReceived` event.
+        """
+        self.__xbee_device.del_socket_state_received_callback(callback)
+
+    def get_sock_info(self):
+        """
+        Returns the information of this socket.
+
+        Returns:
+            :class:`.SocketInfo`: The socket information.
+
+        Raises:
+            InvalidOperatingModeException: if the XBee device's operating mode is not API or ESCAPED API. This
+                method only checks the cached value of the operating mode.
+            TimeoutException: if the response is not received before the read timeout expires.
+            XBeeException: if the XBee device's serial port is closed.
+
+        .. seealso::
+           | :class:`.SocketInfo`
+        """
+        return self.__xbee_device.get_socket_info(self.__socket_id)
+
+    def __create_socket(self):
+        """
+        Creates a new socket by sending a :class:`.SocketCreatePacket`.
+
+        Raises:
+            TimeoutException: if the response is not received in the configured timeout.
+            XBeeSocketException: if the response contains any error.
+        """
+        # Create, send and check the socket create packet.
+        create_packet = SocketCreatePacket(self.__xbee_device.get_next_frame_id(), self.__ip_protocol)
+        response_packet = self.__xbee_device.send_packet_sync_and_get_response(create_packet, self.__timeout)
+        self.__check_response(response_packet)
+
+        # Store the received socket ID.
+        self.__socket_id = response_packet.socket_id
+
+    def __register_state_callback(self):
+        """
+        Registers the socket state callback to be notified when an error occurs.
+        """
+        if self.__socket_state_callback:
+            return
+
+        def socket_state_callback(socket_id, state):
+            if socket_id != socket_id:
```

Comment:
I guess this is comparing the same variable, hence it is never going to enter through the return.
`if self.__socket_id != socket_id:`